rust/sdk-processor/src/steps/parquet_default_processor/parquet_default_extractor.rs

```diff
@@ -0,0 +1,125 @@
+use crate::{
+    parquet_processors::{ParquetTypeEnum, ParquetTypeStructs},
+    utils::parquet_extractor_helper::add_to_map_if_opted_in_for_backfill,
+};
+use aptos_indexer_processor_sdk::{
+    aptos_protos::transaction::v1::Transaction,
+    traits::{async_step::AsyncRunType, AsyncStep, NamedStep, Processable},
+    types::transaction_context::TransactionContext,
+    utils::errors::ProcessorError,
+};
+use async_trait::async_trait;
+use processor::{
+    db::common::models::default_models::{
+        move_tables::TableItemConvertible, parquet_move_tables::TableItem,
+    },
+    processors::{
+        default_processor::process_transactions,
+        parquet_processors::parquet_default_processor::process_transactions_parquet,
+    },
+    worker::TableFlags,
+};
+use std::collections::HashMap;
+use tracing::debug;
+
+/// Extracts parquet data from transactions, allowing optional selection of specific tables.
+pub struct ParquetDefaultExtractor
+where
+    Self: Processable + Send + Sized + 'static,
+{
+    pub opt_in_tables: TableFlags,
+}
+
+type ParquetTypeMap = HashMap<ParquetTypeEnum, ParquetTypeStructs>;
+
+#[async_trait]
+impl Processable for ParquetDefaultExtractor {
+    type Input = Vec<Transaction>;
+    type Output = ParquetTypeMap;
+    type RunType = AsyncRunType;
+
+    async fn process(
+        &mut self,
+        transactions: TransactionContext<Self::Input>,
+    ) -> anyhow::Result<Option<TransactionContext<ParquetTypeMap>>, ProcessorError> {
+        let (_, raw_table_items, _, _) =
+            process_transactions(transactions.data.clone(), TableFlags::empty());
+
+        let parquet_table_items: Vec<TableItem> =
+            raw_table_items.iter().map(TableItem::from_raw).collect();
+
+        let (
+            (
+                move_resources,
+                write_set_changes,
+                parquet_transactions,
+                _parquet_table_items,
+                move_modules,
+            ),
+            _transaction_version_to_struct_count,
```

Comment:
curious what the convention is in general for var names with leading underscore?

Reply:
It means the variables are unused, so the compiler doesn't mark this as a lint error

Reply:
^ +1

Reply:
when would you do the leading underscore instead of just the underscore like in 

```
 let (txns, _, write_set_changes, wsc_details) = TransactionModel::from_transactions(
        &transactions,
        &mut transaction_version_to_struct_count,
    );
```